parse.py

```diff
@@ -0,0 +1,292 @@
+#-*- coding: utf-8 -*-
+
+import html5, string
+
+# Global variables
+_tags = None
+_domParser = None
+
+def _convertEncodedText(txt):
+	"""
+	Convert HTML-encoded text into decoded string.
+
+	The reason for this function is the handling of HTML entities, which is not
+	properly supported by native JavaScript.
+
+	We use the browser's DOM parser to to this, according to
+	https://stackoverflow.com/questions/3700326/decode-amp-back-to-in-javascript
+
+	:param txt: The encoded text.
+	:return: The decoded text.
+	"""
+	global _domParser
+
+	if _domParser is None:
+		_domParser = eval("new DOMParser")
+
+	dom = _domParser.parseFromString("<!doctype html><body>" + str(txt), "text/html")
+	return dom.body.textContent
+
+def _buildDescription():
+	"""
+	Generates a dictionary of all to the html5-library
+	known tags and their associated objects and attributes.
+	"""
+	tags = {}
+
+	for cname in dir(html5):
+		cl = getattr(html5, cname)
+		try:
+
+			inst = cl()
```

Comment:
For security, don't instanciate each class optimistically and then use isinstance, check if cl issubclass of widget. (Prevents you from accidental calling each function that might exist in that namespace

Reply:
OK I've done a fix for this which is committed soon.